app/models/document.rb

```diff
@@ -71,4 +71,8 @@ def document_topics
     @document_topics_index ||= TopicIndexService.new
     DocumentTopics.find_by_document(self, @document_topics_index)
   end
+
+  def retirement
+    self.timeline_entries.where(entry_type: "retired").order(:created_at).last&.retirement
```

Comment:
I think we should find a way to associate with a retirement without using timeline entry.

Timeline entry's role should be just around the timeline and the document should be able to know it's own state without consulting it.

Reply:
This should be addressed in this commit https://github.com/alphagov/content-publisher/pull/608/commits/82bf51135c41cec2d7ef58880cbfb2dac56fca21